ext/CatalystStructuralIdentifiabilityExtension/structural_identifiability_extension.jl

```diff
@@ -0,0 +1,144 @@
+                                        known_p = Num[], remove_conserved = true, ignore_no_measured_warn=false, 
+                                        kwargs...)
+    # Creates a ODESystem, and list of measured quantities, of SI's preferred form.
+    osys, conseqs = make_osys(rs; remove_conserved)
+    measured_quantities = make_measured_quantities(rs, measured_quantities, known_p, conseqs; ignore_no_measured_warn)
+
+    # Computes identifiable functions and converts it to a easy to read form.
+    out = SI.find_identifiable_functions(osys, args...; measured_quantities, kwargs...)
+    return vector_subs(out, reverse.(conseqs))
+end
+
+### Helper Functions ###
+
+# From a reaction system, creates the corresponding MTK-style ODESystem for SI application 
+# Also compute the, later needed, conservation law equations and list of system symbols (states and parameters).
+function make_osys(rs::ReactionSystem; remove_conserved=true)
+    # Creates the ODESystem corresponding to the ReactionSystem (expanding functions and flattening it).
+    # Creates a list of the systems all symbols (states and parameters).
+    rs = Catalyst.expand_registered_functions(flatten(rs))
+    osys = convert(ODESystem, rs; remove_conserved)
+    vars = [states(rs); parameters(rs)]
+
+    # Computes equations for system conservation laws.
+    # These cannot be computed for hierarchical systems (and hence this is skipped). 
+    # If there are no conserved equations, the `conseqs` variable must still have the `Vector{Pair{Any, Any}}` type.
+    if !remove_conserved
+        conseqs = Vector{Pair{Any, Any}}[]
+    else
+        conseqs = [ceq.lhs => ceq.rhs for ceq in conservedequations(rs)]
+        isempty(conseqs) && (conseqs = Vector{Pair{Any, Any}}[])
+    end
+
+    return osys, conseqs, vars
+end
+
+# Creates a list of measured quantities of a form that SI can read.
+# Each measured quantity must have a form like:
+# `obs_var ~ X` # (Here, `obs_var` is a variable, and X is whatever we can measure).
+function make_measured_quantities(rs::ReactionSystem, measured_quantities::Vector{T}, known_p::Vector{S},
+                                  conseqs; ignore_no_measured_warn=false) where {T,S}
+    # Warning if the user didn't give any measured quantities.
+    if ignore_no_measured_warn || isempty(measured_quantities) 
+        @warn "No measured quantity provided to the `measured_quantities` argument, any further identifiability analysis will likely fail. You can disable this warning by setting `ignore_no_measured_warn=true`."
+    end
+
+    # Appends the known parameters to the measured_quantities vector. Converts any Symbols to symbolics.
+    measured_quantities = [measured_quantities; known_p]
+    measured_quantities = [(q isa Symbol) ? Catalyst._symbol_to_var(rs, q) : q for q in measured_quantities]
+    measured_quantities = vector_subs(measured_quantities, conseqs)
```

Comment:
```suggestion
    mqiterator = Iterators.flatten((measured_quantities, known_p))
    mqs = [(q isa Symbol) ? Catalyst._symbol_to_var(rs, q) : q for q in mqiterator]
    mqs = vector_subs(measured_quantities, conseqs)
```
This avoids potentially repeatedly changing the type of `measured_quantities`, and I think is more readable.